packaging/gardener/cmd/gardener-extension-cri-rm/main.go

```diff
@@ -0,0 +1,286 @@
+				IgnoreOperationAnnotation: ignoreOperationAnnotation,
+			}); err != nil {
+				return fmt.Errorf("error configuring actuator: %s", err)
+			}
+
+			if err := mgr.Start(ctx); err != nil {
+				return fmt.Errorf("error running manager: %s", err)
+			}
+
+			return nil
+		},
+	}
+
+	optionAggregator.AddFlags(cmd.Flags())
+
+	if err := cmd.ExecuteContext(ctx); err != nil {
+		controllercmd.LogErrAndExit(err, "error executing the main controller command")
+	}
+
+}
+
+// ---------------------------------------------------------------------------------------
+// -                                        Actuator                                     -
+// ---------------------------------------------------------------------------------------
+
+func NewActuator() extension.Actuator {
+	return &actuator{
+		chartRendererFactory: extensionscontroller.ChartRendererFactoryFunc(util.NewChartRendererForShoot),
+		logger:               log.Log.WithName(ActuatorName),
+	}
+}
+
+type actuator struct {
+	client               client.Client
+	config               *rest.Config
+	chartRendererFactory extensionscontroller.ChartRendererFactory
+	decoder              runtime.Decoder
+	logger               logr.Logger
+}
+
+func (a *actuator) Reconcile(ctx context.Context, ex *extensionsv1alpha1.Extension) error {
+
+	// Find what shoot cluster we dealing with.
+	namespace := ex.GetNamespace()
+	cluster, err := controller.GetCluster(ctx, a.client, namespace)
+	if err != nil {
+		return err
+	}
+	a.logger.Info("Reconcile: checking extension...") // , "shoot", cluster.Shoot.Name, "namespace", cluster.Shoot.Namespace)
+
+	mr := &resourcemanagerv1alpha1.ManagedResource{}
+	if err := a.client.Get(ctx, kutil.Key(namespace, ManagedResourceName), mr); err != nil {
+		// Continue only if not found.
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+
+		a.logger.Info("Reconcile: installing extension (managedresource)...") // , "shoot", cluster.Shoot.Name, "namespace", cluster.Shoot.Namespace)
+		// Depending on shoot, chartredner will have different capabilities based on K8s version..
+		chartRenderer, err := a.chartRendererFactory.NewChartRendererForShoot(cluster.Shoot.Spec.Kubernetes.Version)
+
+		chartValues := map[string]interface{}{
+			"images": map[string]string{
+				InstallationImageName: "foo", // TODO: imagevector.FindImage(InstallationImageName),
+			},
+		}
+
+		release, err := chartRenderer.Render(ChartPath, InstallationReleaseName, metav1.NamespaceSystem, chartValues)
+		if err != nil {
+			return err
+		}
+
+		// Put chart into secret
+		secretData := map[string][]byte{ConfigKey: release.Manifest()}
+
+		// Reconcile managedresource and secret for shoot.
+		return managedresources.CreateForShoot(ctx, a.client, namespace, ManagedResourceName, false, secretData)
+	} else {
+		a.logger.Info("Reconcile: extension is already installed. Ignoring.") //, "shoot", cluster.Shoot.Name, "namespace", cluster.Shoot.Namespace)
```

Comment:
You probably want to always deploy the managed resource (think extension upgrade with changes in your resources/charts)